plugins/inputs/stackdriver/README.md

```diff
@@ -0,0 +1,100 @@
+# Stackdriver Input Plugin
+
+Stackdriver scrapes metrics from Google Cloud Monitoring.
+
+### Credentials
+
+This plugin uses a [Google Service Account](https://cloud.google.com/docs/authentication/getting-started) to interacting with the Stackdriver Monitoring API.
+
+### Configuration:
+
+```toml
+[[inputs.stackdriver]]
+  ## GCP Project
+  project = "projects/erudite-bloom-151019"
+  
+  ## API rate limit. On a default project, it seems that a single user can make
+  ## ~14 requests per second. This might be configurable. Each API request can
+  ## fetch every time series for a single metric type, though -- this is plenty
+  ## fast for scraping all the builtin metric types (and even a handful of
+  ## custom ones) every 60s.
+  # rateLimit = 14
+  
+  ## Every query to stackdriver queries for data points that have timestamp t
+  ## such that: (now() - lookbackSeconds) <= t < now(). Note that influx will
+  ## de-dedupe points that are pulled twice, so it's best to be safe here, just
+  ## in case it takes GCP awhile to get around to recording the data you seek.
+  # lookbackSeconds = 600
+  
+  ## Collection Delay Seconds (required - must account for metrics availability via Stackdriver Monitoring API)
+  # delaySeconds = 60
+  
+  ## Recommended: use metric 'interval' that equals to 'lookbackSeconds' to avoid
+  ## gaps or overlap in pulled data
```

Comment:
Hi, it's great to see this plugin get more traction now!

Why is it bad to have overlap? InfluxDB de-dupes points that re pulled twice, as is describe in the comments for `lookbackSeconds`. Based on that comment it sounds like metrics come in late sometimes (`delaySeconds`), so it can be good to have overlap to fill in gaps. Does your experience show otherwise?

Reply:
Thanks for your advice. The comment is updated.

Reply:
Thanks! The updated comment make sense to me.

Reply:
I think the comments are still misleading. Based on https://github.com/influxdata/telegraf/pull/4977/files#diff-7f20909bbc5d4389bc3295b1477ec408R288 the window that data is pulled from is not

`(now() - lookbackSeconds) <= t < now()`

but rather `(now() - lookbackSeconds) <= t < (now() - delaySeconds)`

right?

Won't the way this is calculated leave a gap with missing data?

Imagine i have the defaults of

```
delaySeconds = 60
lookbackseconds == 600
interval == 10m
```
That means if i start telegraf 

at 2:00:00 it will pull data from 1:50:00 to 1:59:00
at 2:10:00 it will pull data from 2:00:00 to 2:09:00

and I missed the data from 1:59:00 to 2:00:00.

Might the fix be to change the window to 

`(now() - lookbackSeconds - delaySeconds) <= t < (now() - delaySeconds)`

?

If you go with that, consider renaming lookbackSeconds to something like windowWidth.